ement-room.el

```diff
@@ -4449,21 +4453,50 @@ is non-nil."
   ;; The following may also clear the cache in order to force a recalculation:
   ;; - `ement-room-compose-buffer-window-state-change-handler'
   ;; - `ement-room-compose-buffer-window-buffer-change-handler'
+  ;;
+  ;; Global mutex `ement-room-compose-buffer-window-auto-height-resizing-p'
+  ;; ensures that we cannot run recursively.  We also resize only the selected
+  ;; window, even if there are compose buffers displayed in other windows which
+  ;; might also be affected.  This conservative approach can prevent desirable
+  ;; resizing in some cases, but restricting our behaviour this way keeps things
+  ;; simple so that we needn't consider potential issues such as endless cycles
+  ;; of conflicting resizes.
+  ;;
+  ;; Perfection would in any case be non-trivial -- consider two compose windows
+  ;; side-by-side in a horizontal split, each showing a different compose buffer
+  ;; with a different desired height.  We cannot have the "correct" size for
+  ;; both simultaneously.  The best thing to do would be to maintain the tallest
+  ;; height amongst all conflicting windows at all times -- but that is, again,
+  ;; considerably more complex.
+  ;;
+  ;; Most of the time the window arrangements are expected to be very simple and
+  ;; so a more comprehensive solution, while possible, is not worth the added
+  ;; complexity -- our relatively simplistic approach is good enough for the
+  ;; vast majority of situations.
+
+  ;; Skip resizing if we are being called recursively...
   (unless (or (bound-and-true-p ement-room-compose-buffer-window-auto-height-resizing-p)
-              (window-full-height-p))
+              ;; ...or there are no other windows to resize...
+              (window-full-height-p)
+              ;; ...or we have just switched to this buffer from another buffer
+              ;; (we may be cycling window buffers, and about to switch again).
+              (and (window-old-buffer)
+                   (not (eq (window-old-buffer) (current-buffer)))))
     ;; Manipulate the window body height.
     (let* ((pixelwise (and ement-room-compose-buffer-window-auto-height-pixelwise
                            (display-graphic-p)))
            (lineheight (and pixelwise (default-line-height)))
            (buflines (max 1 (count-screen-lines nil nil t)))
            (cache (if pixelwise
                       (* buflines lineheight)
-                    buflines)))
+                    buflines))
+           (wcache (window-parameter
+                    nil 'ement-room-compose-buffer-window-auto-height-cache)))
       ;; Do nothing if the desired height has not changed.
-      (unless (and (boundp 'ement-room-compose-buffer-window-auto-height-cache)
-                   (eql cache ement-room-compose-buffer-window-auto-height-cache))
+      (unless (and wcache (eql cache wcache))
```

Comment:
An extreme nitpick, probably, but also just a curiosity for me: Why use `eql` instead of `eq` or `equal`?  (AFAIR the only reason to use `eql` is for certain numerical comparisons.)  It might be clearer (to the reader) to use one or the other instead of `eql`.

Reply:
The cache valued should always be either an integer or `nil`, so `eql` seemed best (`=` signals an error for `nil`).  It's a bit of added type-checking and also a visual indication that we're dealing with numbers.

Reply:
Thanks, I learned something today.  :)